Assets/Scripts/BulletManagement/SlipTimeEmitter.cs

```diff
@@ -11,14 +11,100 @@ public class SlipTimeEmitter : EmitterBase, ISlipTimeAdherent
         [SerializeField]
         private SlipTimeManager slipTimeManager;
         
+        // Running total of emitted bullet waves
+        private int cyclesEmitted = 0;
+
+        private bool limited = false;
+
+        public float bulletSpeedMultiplier = 1.0f;
+        
+        // The number of bullet waves to emit. Set to zero for no limit.
+        public int emitCycles = 0;
+
+        // Number of bullets to emit each emit cycle.
+        public int emitBulletCount;
+        
+        // Angle at which the emtiter points
+        public float startAngle = 0f;
+
+        // The rate at which the emitter should spin each update cycle
+        public float spinRate = 0f;
+
+        // The maximum spin rate at which spin will reverse
+        public float spinMax = 1f;
+        
+        // Analogous to angular acceleration, the rate at which spinRate increases
+        public float spinAccel = 0f;
+        
+        // Flag used to control if the spin should reverse upon hitting max speed
+        public bool reverseSpin = false;
+        
+        // Determines if the emitter should aim at the player or not. Note that this will overwrite startAngle.
+        public bool aimed = false;
+
+        void Start()
+        {
+            limited = emitCycles != 0;
+            this.bulletPattern = this.GetComponent<BulletPattern>();
+            if (aimed)
+            {
+                var angleToPlayer = AngleToPlayer();
+                transform.Rotate(Vector3.forward * -angleToPlayer, Space.Self);
+            }
+            else
+            {
+                transform.Rotate(Vector3.forward * startAngle, Space.Self);
+            }
+
+        }
+        
+        protected float AngleToPlayer()
+        {
+            var playerTransform = GameObject.Find("Player").transform;
+            var emitterTransform = transform;
+            Vector3 targetDir = playerTransform.position - emitterTransform.position;
+            return Vector3.SignedAngle(targetDir, emitterTransform.up, Vector3.forward);
+        }
+
+        private void EmitBullets()
```

Comment:
I think latest MR has this moved into the base class? I think we can remove this.

Reply:
Correct me if I'm wrong, but I believe the base class' implementation uses `Quaternion.Identity` in the `Instantiate` call. Would you mind if I changed it to `currentTransform.rotation` in the base class and then got rid of this method here?

Reply:
Yep! I think that would reduce maintenance overhead over time.